plugins/services/src/js/reducers/serviceForm/Container.js

```diff
@@ -179,6 +191,15 @@ module.exports = {
       delete newState.docker;
     } else if (ValidatorUtil.isEmpty(newState.docker.image)) {
```

Comment:
Instead of all the else if's - maybe we could do something like:
```
const clearDocker = ValidatorUtil.isEmpty(newState.docker) ||
  ValidatorUtil.isEmpty(newState.docker.image) ||
  ValidatorUtil.isEmpty(newState.type) ||
  (this.isMesosRuntime && !ValidatorUtil.isEmpty(newState.docker));

if (clearDocker) {
  delete newState.docker;
}
```